packages/coordinator/ts/proof/proof.service.ts

```diff
@@ -0,0 +1,347 @@
+      abi: MACIFactory.abi,
+      functionName: "getPoll",
+      args: [BigInt(pollId)],
+    });
+
+    const pollAddress = pollContracts.poll;
+    if (pollAddress.toLowerCase() === ZeroAddress.toLowerCase()) {
+      this.logger.error(`Error: ${ErrorCodes.POLL_NOT_FOUND}, Poll ${pollId} not found`);
+      throw new Error(ErrorCodes.POLL_NOT_FOUND.toString());
+    }
+
+    // check if state tree has been merged
+    const isStateMerged = await publicClient.readContract({
+      address: pollAddress,
+      abi: PollFactory.abi,
+      functionName: "stateMerged",
+    });
+    if (!isStateMerged) {
+      this.logger.error(`Error: ${ErrorCodes.NOT_MERGED_STATE_TREE}, state tree is not merged`);
+      throw new Error(ErrorCodes.NOT_MERGED_STATE_TREE.toString());
+    }
+
+    const [maciContract, mpContract, pollContract, tallyContract, vkRegistryContract, verifierContract] =
+      await Promise.all([
+        this.deployment.getContract<MACI>({
+          name: EContracts.MACI,
+          address: maciContractAddress,
+        }),
+        this.deployment.getContract<MessageProcessor>({
+          name: EContracts.MessageProcessor,
+          address: pollContracts.messageProcessor,
+        }),
+        this.deployment.getContract<Poll>({
+          name: EContracts.Poll,
+          address: pollContracts.poll,
+        }),
+        this.deployment.getContract<Tally>({
+          name: EContracts.Tally,
+          address: pollContracts.tally,
+        }),
+        this.deployment.getContract<VkRegistry>({ name: EContracts.VkRegistry }),
+        this.deployment.getContract<Verifier>({ name: EContracts.Verifier }),
+      ]);
+    const prover = new Prover({
+      maciContract,
+      mpContract,
+      pollContract,
+      tallyContract,
+      vkRegistryContract,
+      verifierContract,
+    });
+
+    const data = {
+      processProofs: await this.readProofs(path.resolve("./proofs"), "process"),
+      tallyProofs: await this.readProofs(path.resolve("./proofs"), "tally"),
+    };
+    // prove message processing
+    await prover.proveMessageProcessing(data.processProofs);
+    // read tally data
+    const tallyData = await fs.promises
+      .readFile("./tally.json", "utf8")
+      .then((result) => JSON.parse(result) as unknown as TallyData);
+    // prove tally
+    await prover.proveTally(data.tallyProofs);
+    // get number of participants from the registry
+    const numSignUpsAndMessages = await pollContract.numSignUpsAndMessages();
+    const numSignUps = numSignUpsAndMessages[0];
+    /* const registryContractAddress = await pollContract.getRegistry();
+    const registryContract = SimpleRegistryFactory.connect(registryContractAddress, signer);
+    const recipientCount = await registryContract.recipientCount();
+    await prover.submitResults(tallyData, Number.parseInt(recipientCount.toString(), 10));
+    */
+    // submit results
+    await prover.submitResults(tallyData, Number.parseInt(numSignUps.toString(), 10));
```

Comment:
that would not work cause num signups can be way more than vote options. use this variable https://github.com/privacy-scaling-explorations/maci/blob/dev/packages/contracts/contracts/Poll.sol#L179